pulser/parametrized/variable.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2020 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Union
+
+import numpy as np
+
+from pulser.parametrized import Parametrized
+from pulser.parametrized.paramobj import OpSupport
+
+
+@dataclass(frozen=True, eq=False)
+class Variable(Parametrized, OpSupport):
+    """A variable for parametrized sequence building.
+
+    Args:
+        name (str): Unique name for the variable.
+        dtype (type): Type of the variable's content. Supports `float`, `int`
+            and `float`.
+        size (int=1): The number of values stored. Defaults to a single value.
+    """
+
+    name: str
+    dtype: type
+    size: int = 1
+
+    def __post_init__(self):
+        if not isinstance(self.name, str):
+            raise TypeError("Variable's 'name' has to be of type 'str'.")
+        if self.dtype not in [int, float, str]:
+            raise TypeError(f"Invalid data type '{self.dtype}' for Variable.")
+        if not isinstance(self.size, int):
+            raise TypeError("Given variable 'size' is not of type 'int'.")
+        elif self.size < 1:
+            raise ValueError("Variables must be of size 1 or larger.")
+        self.__dict__["_count"] = -1      # Counts the updates
+        self._clear()
+
+    @property
+    def variables(self):
+        return {self.name: self}
+
+    def _clear(self):
+        self.__dict__["value"] = None
+        self.__dict__["_count"] += 1
+
+    def _assign(self, value):
+        if self.dtype == str:
```

Comment:
For the other dtypes, I didn't want to be picky with whether the user gave an `int` when it was supposed to be a `float`, so I'm using line 66:
```
val = np.array(value, dtype=self.dtype)
```
to cast the given value (or values) to the correct dtype. I figured that if it was not possible, this call would throw the error. 
The reason I don't do this for `str` is because you can cast anything to `str`.